store/models/address.go

```diff
@@ -0,0 +1,79 @@
+package models
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	addressT = reflect.TypeOf(EIP55Address(""))
+)
+
+// EIP55Address is a newtype for string which persists an ethereum address in
+// its original string representation which includes a leading 0x, and EIP55
+// checksum which is represented by the case of digits A-F.
+type EIP55Address string
+
+// Str returns the underlying string representation.
+func (a EIP55Address) Str() string { return a.Address().Str() }
+
+// Bytes returns the raw bytes
+func (a EIP55Address) Bytes() []byte { return a.Address().Bytes() }
+
+// Big returns a big.Int representation
+func (a EIP55Address) Big() *big.Int { return a.Address().Big() }
+
+// Hash returns the Hash
+func (a EIP55Address) Hash() common.Hash { return a.Address().Hash() }
+
+// Address returns EIP55Address as a go-ethereum Address type
+func (a EIP55Address) Address() common.Address { return common.HexToAddress(a.String()) }
```

Comment:
Shouldn't this be `return common.HexToAddress(a.Hex())` so we leverage the checksum, or do we want it to be verbatim to prevent unintended manipulation?

Reply:
`String` should always return the EIP55 representation now, except in the case of the zero value, which is difficult to avoid :/